cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -490,6 +527,29 @@ public ClusterModel clusterModel(long from,
           clusterModel.setBrokerState(brokerId, Broker.State.BAD_DISKS);
         }
       }
+
+      // Populate disk information for the cluster model.
+      if (populateDiskInfo) {
+        Map<Integer, KafkaFuture<Map<String, LogDirInfo>>> logDirsByBrokerId =
+            _adminClient.describeLogDirs(clusterModel.aliveBrokers().stream().mapToInt(Broker::id).boxed().collect(Collectors.toList())).values();
+        for (Map.Entry<Integer, KafkaFuture<Map<String, LogDirInfo>>> entry : logDirsByBrokerId.entrySet()) {
+          Broker broker = clusterModel.broker(entry.getKey());
+          try {
+            entry.getValue().get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((key, value) -> {
+              if (value.error == Errors.NONE) {
+                broker.populateDiskInfo(key, value.replicaInfos);
+              } else {
+                broker.disk(key).setState(DEAD);
+              }
+            });
+          } catch (TimeoutException te) {
+            LOG.error("Getting logdir information for broker {} timed out.", entry.getKey());
+          } catch (InterruptedException | ExecutionException e) {
+            LOG.error("Populating logdir information for broker {} encountered Exception {}.", entry.getKey(), e);
```

Comment:
If we cannot populate disk info for some reason, then do we really want to just log the error and continue (current behavior), or return this failure to the user and terminate the process? IIUC, `populateDiskInfo` is set to `true` only when the disk-related goals are required for completion of the process.

Can you clarify what the expected behavior should be?

Reply:
My original thought is if we fail to get logdir Information here for certain broker, disk-related goals will do nothing for this broker, but it should be able to work on the broker which has logdir information returned. So I did not thrown exception and terminate the operation here.

As you can see from this patch, there are several place I need to handle response for `DescribeLogDir` and `AlterLogdir`.
1) generate cluster model(here)
2) generate `ExecutionTask` from `ExecutionProposal`
3) execute generated task
4) check the task finish or not

My basic handling principle is that 
1) first priority is safety, I do not want to trigger a change-state operation(i.e. `AlterLogdir`) unless I collect enough information and am sure that it helps to make cluster more balanced
2) second priority is availability, I am aware I may get some ERROR_RESPONSE for some broker in the cluster, but it should not block me from optimizing other broker in the cluster.

So my handling for the above 4 place is
1) generate cluster model(here)  -> ignore this broker and not generate any proposal for this broker
2) generate `ExecutionTask` from `ExecutionProposal` -> do not generate task from proposal
3) execute the task -> mark task as aborted
4) check the task finish or not -> retry